apache/superset
# fix: avoid 500 errors with SQLLAB_BACKEND_PERSISTENCE

### SUMMARY


There are cases when SqlLab generates 500 errors on backend due to foreignkey violation

```
sqlalchemy.exc.IntegrityError: (pymysql.err.IntegrityError) (1452, 'Cannot add or update a child row: a foreign key constraint fails (`...`.`tab_state`, CONSTRAINT `tab_state_latest_query_id_fkey` FOREIGN KEY (`latest_query_id`) REFERENCES `query` (`client_id`) ON DELETE SET NULL)')
```

Maybe some front-end logic should be updated for this scenario too but the backend shouldn't crash with 500 in such cases anyway.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS


SQLLAB_BACKEND_PERSISTENCE should be enabled.

1. Open SqlLab and write a query.
2. Turn off network.
3. Press Run to try to run SQL query.
4. Turn on network and wait a bit.

Expected result:
Errors are fine, but there shouldn't be 500 errors on backend.

### ADDITIONAL INFORMATION
